### PR TITLE
Fix memory leaks with physics

### PR DESCRIPTION
**Pull request by andens**

### CODE REVIEW

File: src/Engine/Manager/PhysicsManager.cpp [hunk 1]
```diff
@@ -42,6 +42,10 @@ PhysicsManager::~PhysicsManager() {
     delete dispatcher;
     delete collisionConfiguration;
     delete broadphase;
+
+    for (auto t : triggers) {
```
Comment (Chainsawkitten) [inline src/Engine/Manager/PhysicsManager.cpp]: No brackets on single-line for.

File: src/Engine/Physics/Trigger.cpp [hunk 2]
```diff
@@ -7,17 +7,17 @@
 
 namespace Physics {
     Trigger::Trigger(const btTransform& transform) {
-        trigger = new btCollisionObject();
+        trigger.reset(new btCollisionObject());
         trigger->setWorldTransform(transform);
     }
 
     btCollisionObject* Trigger::GetCollisionObject() const {
```
Comment (NinjaDanz3r) [inline src/Engine/Physics/Trigger.cpp]: MINOR The function 'GetCollisionObject' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**